sdk/core/azure-core/src/http/url.cpp

```diff
@@ -49,7 +49,7 @@ Url::Url(const std::string& url)
   if (pos != url.end() && *pos == '?')
   {
     auto queryIter = std::find(pos + 1, url.end(), '#');
-    AppendQueries(std::string(pos + 1, queryIter));
+    AppendEncodedQueries(std::string(pos + 1, queryIter));
```

Comment:
Should we `Encode` the `std::string(pos + 1, queryIter)` before passing it to `AppendEncodedQueries` or is that already considered encoded?

Reply:
query strings usually appear in encoded form.

Reply:
Usually, or always? If it is `usually`, then the behavior will be unexpected in some cases. I think the contract should be clear on what data we expect.

Reply:
From Storage's perspective, we usually pass encoded full URL. I don't know if there are any other situations core team will need to handle.